### PR TITLE
Add support for nan in RESP3 double

### LINKED ISSUE #9: Add support for `nan` in RESP3 double
Currenlty RESP3 support 2 "special values": `inf` and `-inf`:
```
,inf<CR><LF>
,-inf<CR><LF>
```

I think we should add a third one - `nan`:
```
,nan<CR><LF>
```

This is already being used (even though it's not in the spec) in [TDIGEST.CDF](https://redis.io/commands/tdigest.cdf/)

### PR DESCRIPTION
**Pull request by filipecosta90**
Fixes #9

### CODE REVIEW

**Comment by oranagra** [comment]
This was conceptually approved in a core-team meeting.
@filipecosta90 can you please extend it to mention negative NaN as well?

**Comment by filipecosta90** [comment]
> This was conceptually approved in a core-team meeting. @filipecosta90 can you please extend it to mention negative NaN as well?

sure. doing it right now :)

**Comment by filipecosta90** [comment]
@oranagra done. please confirm all is OK

**Review by oranagra — APPROVED** [review]

**Comment by leibale** [comment]
Negative "NaN" (not a number)? what's the difference between "+NaN" and "-NaN"?

**Comment by oranagra** [comment]
don't know, but evidently (see https://github.com/redis/redis/pull/11482) it has a sign.. 
wikipedia says so too.. didn't bother to look much into it.

**Comment by filipecosta90** [comment]
@leibale and @oranagra according to this [answer](https://stackoverflow.com/questions/21349847/positive-vs-negative-nans) and the IEEE IEEE:

>  Conversion of a quiet NaN in a supported format to an external character sequence shall produce a language-defined one of “nan” or a sequence that is equivalent except for case (e.g., “NaN”), with an optional preceding sign. **(This standard does not interpret the sign of a NaN.)**

meaning that NaN can be represented by any pattern of and sometimes printed.

> A signalling NaN is represented by any bit pattern
between 7FF0000000000001 and 7FF7FFFFFFFFFFFF or
between FFF0000000000001 and FFF7FFFFFFFFFFFF

TLDR `-nan`and `nan` should be valid and equivalent and interpreted the same by any RESP client.  agree?

**Comment by oranagra** [comment]
agree.. so the code in both PRs (test and docs) is good to be merged, right?
